src/overlays/actors/ovl_Bg_Tobira01/z_bg_tobira01.c

```diff
@@ -21,14 +20,87 @@ const ActorInit Bg_Tobira01_InitVars = {
     (ActorFunc)BgTobira01_Update,
     (ActorFunc)BgTobira01_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Tobira01_0x80B12430/func_80B12430.asm")
+extern Gfx D_06000088[];
+extern CollisionHeader D_060011C0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Tobira01_0x80B12430/BgTobira01_Init.asm")
+void BgTobira01_Open(BgTobira01* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    s16 cutsceneId = this->dyna.actor.cutscene;
+    s16 prevTimer;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Tobira01_0x80B12430/BgTobira01_Destroy.asm")
+    if (this->playCutscene) {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+        } else if (ActorCutscene_GetCanPlayNext(cutsceneId) != 0) {
+            ActorCutscene_StartAndSetUnkLinkFields(cutsceneId, &this->dyna.actor);
+            gSaveContext.weekEventReg[88] |= 0x40;
+            this->playCutscene = false;
+        } else {
+            ActorCutscene_SetIntentToPlay(cutsceneId);
+        }
+    } else if (!(gSaveContext.weekEventReg[88] & 0x40) && (this->timer == 0) && (globalCtx->actorCtx.unk1F5 != 0) &&
+               (globalCtx->actorCtx.unk1F4 == 0) &&
+               (func_800C99AC(&globalCtx->colCtx, player->base.floorPoly, player->base.floorBgId) == 6)) {
+        this->playCutscene = true;
+        this->unk_16C = 0; // this variable is not used anywhere else
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Tobira01_0x80B12430/BgTobira01_Update.asm")
+    prevTimer = this->timer;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Bg_Tobira01_0x80B12430/BgTobira01_Draw.asm")
+    if (gSaveContext.weekEventReg[88] & 0x40) {
+        this->timer++;
+    } else {
+        this->timer--;
+    };
+
+    this->timer = CLAMP(this->timer, 0, 60);
+
+    if (this->timer != prevTimer) {
+        if (1) {}
+        Audio_PlayActorSound2(&this->dyna.actor, 0x2143);
+        this->dyna.actor.world.pos.y = (this->yOffset = (this->timer * 1.6666666f) + this->dyna.actor.home.pos.y);
+        this->timer2 = 180;
+    }
+
+    if (!(player->stateFlags1 & 0x40) && (gSaveContext.weekEventReg[88] & 0x40) && (DECR(this->timer2) == 0)) {
+        gSaveContext.weekEventReg[88] &= 0xBF;
```

Comment:
```suggestion
        gSaveContext.weekEventReg[88] &= ~0x40;
```
~0x40 might need an explicit cast to (u8)